src/Conjure/Prelude.hs

```diff
@@ -370,50 +372,53 @@ jsonOptions = JSON.defaultOptions
     }
 
 
-class (Functor m, Applicative m, Monad m) => MonadFail m where
-    fail :: Doc -> m a
+class (Functor m, Applicative m, Monad m) => MonadFailDoc m where
```

Comment:
Add MonadFail m to the left-hand side of =>, so we can remove MonadFail m in other places.